services/venice-controller/src/main/java/com/linkedin/venice/controller/server/CreateVersion.java

```diff
@@ -420,9 +420,11 @@ public Route requestTopicForPushing(Admin admin) {
       VersionCreationResponse responseObject = new VersionCreationResponse();
       response.type(HttpConstants.JSON);
       try {
+        boolean missingWriteAccess = !hasWriteAccessToTopic(request);
+        boolean missingReadAccess = this.checkReadMethodForKafka && !hasReadAccessToTopic(request);
+
         // Also allow allowList users to run this command
```

Comment:
Can we rewrite it in a way that `hasReadAccessToTopic` is not call unless it's necessary?

Reply:
Maybe something like (extract error building in private method: `private String buildAclErrorResponse(Response response, Request request, boolean missingWrite, boolean missingRead)`)
```
if (!isAllowListUser(request)) {
  boolean missingWriteAccess = !hasWriteAccessToTopic(request);
  if (missingWriteAccess) {
    return buildAclErrorResponse(response, request, true, false);
  }

  if (this.checkReadMethodForKafka && !hasReadAccessToTopic(request)) {
    return buildAclErrorResponse(response, request, false, true);
  }
}
```

Reply:
Updated, thanks for your suggestion!